frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -0,0 +1,692 @@
+pub struct SignedSubmission<A, B: HasCompact, C> {
+	/// Who submitted this solution.
+	pub(crate) who: A,
+	/// The deposit reserved for storing this solution.
+	pub(crate) deposit: B,
+	/// The reward that should be given to this solution, if chosen the as the final one.
+	pub(crate) reward: B,
+	/// The raw solution itself.
+	pub(crate) solution: RawSolution<C>,
+}
+
+pub(crate) type BalanceOf<T> =
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+pub(crate) type PositiveImbalanceOf<T> = <<T as Config>::Currency as Currency<
+	<T as frame_system::Config>::AccountId,
+>>::PositiveImbalance;
+pub(crate) type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<
+	<T as frame_system::Config>::AccountId,
+>>::NegativeImbalance;
+
+impl<T: Config> Pallet<T> {
+	/// Finish the singed phase. Process the signed submissions from best to worse until a valid one
+	/// is found, rewarding the best one and slashing the invalid ones along the way.
+	///
+	/// Returns true if we have a good solution in the signed phase.
+	///
+	/// This drains the [`SignedSubmissions`], potentially storing the best valid one in
+	/// [`QueuedSolution`].
+	pub fn finalize_signed_phase() -> (bool, Weight) {
+		let mut all_submission: Vec<SignedSubmission<_, _, _>> = <SignedSubmissions<T>>::take();
+		let mut found_solution = false;
+		let mut weight = T::DbWeight::get().reads(1);
+
+		while let Some(best) = all_submission.pop() {
+			let SignedSubmission { solution, who, deposit, reward } = best;
+			let active_voters = solution.compact.voter_count() as u32;
+			let feasibility_weight = {
+				// defensive only: at the end of signed phase, snapshot will exits.
+				let SolutionOrSnapshotSize { voters, targets } =
+					Self::snapshot_metadata().unwrap_or_default();
+				let desired_targets = Self::desired_targets().unwrap_or_default();
+				let v = voters;
+				let t = targets;
+				let a = active_voters;
+				let w = desired_targets;
+				T::WeightInfo::feasibility_check(v, t, a, w)
+			};
+			match Self::feasibility_check(solution, ElectionCompute::Signed) {
+				Ok(ready_solution) => {
+					Self::finalize_signed_phase_accept_solution(
+						ready_solution,
+						&who,
+						deposit,
+						reward,
+					);
+					found_solution = true;
+
+					weight = weight.saturating_add(feasibility_weight);
+					weight = weight
+						.saturating_add(T::WeightInfo::finalize_signed_phase_accept_solution());
+					break;
+				}
+				Err(_) => {
+					// we assume a worse case feasibility check happened anyhow.
+					weight = weight.saturating_add(feasibility_weight);
+					Self::finalize_signed_phase_reject_solution(&who, deposit);
+					weight = weight
+						.saturating_add(T::WeightInfo::finalize_signed_phase_reject_solution());
+				}
+			}
+		}
+
+		// Any unprocessed solution is not pointless to even ponder upon. Feasible or malicious,
+		// they didn't end up being used. Unreserve the bonds.
```

Comment:
```suggestion
		// Any unprocessed solution is pointless to even consider. Feasible or malicious,
		// they didn't end up being used. Unreserve the bonds.
```

Pretty sure the intent got accidentally reversed here.